src/Parser/Shaman/Elemental/Modules/ShamanCore/FrostShock.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import TALENTS from 'common/SPELLS/TALENTS/SHAMAN';
+import SpellLink from 'common/SpellLink';
+import RESOURCE_TYPES from 'common/RESOURCE_TYPES';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import Analyzer from 'Parser/Core/Analyzer';
+
+class FrostShock extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  badFrostShockCount = 0;
+  suboptimalFrostShockCount = 0;  //not used yet, wanna talk to tc-people for thresholds
+
+  on_initialized() {
+    this.active = false;
+  }
+
+  on_byPlayer_cast(event) {
+    if (event.ability.guid !== SPELLS.FROST_SHOCK.id) {
+      return;
+    }
+    this.active = true;
+    if (!this.combatants.selected.hasBuff(TALENTS.ICEFURY_TALENT.id, event.timestamp))
+      this.badFrostShockCount++;
+
+    const resource = event.classResources[0];
```

Comment:
I missed this the first time around, but you should Ideally check if `classResources` is null. You can also use a filter to find the correct resource like this: https://github.com/WoWAnalyzer/WoWAnalyzer/blob/master/src/Parser/Core/Modules/ManaValues.js#L23
Anyway I don't think this should cause problems as Frost Shock always have a resource event afaik, so it's mostly a thing for the future :)